homeassistant/helpers/config_validation.py

```diff
@@ -677,18 +685,32 @@ def __call__(self, data):
                 self.extra = vol.PREVENT_EXTRA
 
             # This is a legacy config, print warning
-            extra_key_errs = [err for err in orig_err.errors
+            extra_key_errs = [err.path[-1] for err in orig_err.errors
                               if err.error_message == 'extra keys not allowed']
             if extra_key_errs:
```

Comment:
We should change this to `if not extra_keys_err: raise orig_err` and then we can drop the `else`

Reply:
Fixed.